internal/reconciler/logpipeline/reconciler.go

```diff
@@ -163,6 +167,17 @@ func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha
 	}
 
 	reconcilablePipelines := r.getReconcilablePipelines(ctx, allPipelines.Items)
+	if len(reconcilablePipelines) == 0 {
+		logf.FromContext(ctx).V(1).Info("cleaning up log pipeline resources: all log pipelines are non-reconcilable")
+		if err = r.deleteResources(ctx); err != nil {
+			return fmt.Errorf("failed to delete log pipeline resources: %w", err)
+		}
+
+		if err = cleanupFinalizersIfNeeded(ctx, r.Client, pipeline); err != nil {
```

Comment:
Think about the consequences of this. Usually finalizers are removed after the deletion of a pipeline once all managed resources have been removed. I'm wondering if removing the finalizers before the pipeline is deleted might have any side-effects in the situation where our operator might not be running.